File: momose/src/parsers/TimeScenarioParser.java
package parsers;

import org.xml.sax.Attributes;

import engine.Scenario;
import engine.SimTime;

public class TimeScenarioParser extends ScenarioParser 
 {
  protected SimTime time;
  
  public TimeScenarioParser()
   {
	super();
	time=null;
   }//Fine costruttore
  
  public Scenario getScenario()
   { return scenario; }
  
  public SimTime getSimTime()
   { return time; }
  
  public void startElement (String uri, String name,String qName, Attributes atts)
   {
	super.startElement(uri, name, qName, atts);  
	
	//Creo il tempo
	if(qName.equals("SimTime"))
	 { createSimTime(atts);  }    
   }//Fine startElement
  
  protected void createSimTime(Attributes atts) 
   {
    //Estraggo gli elementi necessari	 
    float TotalTime=Float.valueOf(atts.getValue(0));
    float dt=Float.valueOf(atts.getValue(1));
    int loops=Integer.valueOf(atts.getValue(2));

    //Creo l'oggetto ScenRectangle
    time=new SimTime(TotalTime,dt,loops);  
   }//Fine create simTime
 }//Fine TimeScenarioParser

File: momosecpp/src/TimeScenarioParser.cpp
#include "TimeScenarioParser.h"

#include <string>

TimeScenarioParser::TimeScenarioParser():ScenarioParser()
 {simTime=NULL;}

TimeScenarioParser::~TimeScenarioParser(){}


void TimeScenarioParser::OnStartElement(const XML_Char* name,const XML_Char** attrs)
 {
   ScenarioParser::OnStartElement(name,attrs);	 
    //Creo il tempo
    string str=xmlCharVectToString(name);	 	 
    if(str=="SimTime")
     { createSimTime(attrs);  } 
 }//Fine OnStartElement
 
 
void TimeScenarioParser::createSimTime(const XML_Char** attrs) 
   {
     //Estraggo gli elementi necessari	 
     string 	 totalStr=getAttrFromName(attrs,"Time");  
     float totalTime=stringToFloat(totalStr);
	   
     string 	 dtStr=getAttrFromName(attrs,"dt");  	   
     float dt=stringToFloat(dtStr);
	   
     string 	 loopsStr=getAttrFromName(attrs,"Loops");	   
     int loops=stringToInt(loopsStr);

    //Creo l'oggetto ScenRectangle
    simTime=new SimTime(totalTime,dt,loops);  
   }//Fine create simTime
 
 
void TimeScenarioParser::OnCharacterData(const XML_Char* data, int len){}
void TimeScenarioParser::OnEndElement(const XML_Char* name){}